core/src/main/java/org/apache/calcite/sql/type/CompositeOperandTypeChecker.java

```diff
@@ -311,6 +324,22 @@ private boolean check(SqlCallBinding callBinding) {
       throw new AssertionError();
     }
   }
+
+  private boolean checkWithOutTypeCoercion(SqlCallBinding callBinding) {
+    if (!callBinding.getValidator().isTypeCoercionEnabled()) {
+      return false;
+    }
+    for (Ord<SqlOperandTypeChecker> ord : Ord.<SqlOperandTypeChecker>zip(allowedRules)) {
```

Comment:
remove the usage of Ord, which is useless. Replace with foreach.

Reply:
Sure.

Reply:
I agree, `Ord` is not necessary here. (Kotlin has `foreachIndexed`, and we'd need `Ord` a lot less if Java had it too.) 

Note that "without" is one word, so the method name should not contain `WithOut`.

Reply:
Thanks Julian, method name renamed to `checkWithoutTypeCoercion`.